netadapter/netconnection.go

```diff
@@ -54,3 +88,19 @@ func (c *NetConnection) NetAddress() *wire.NetAddress {
 func (c *NetConnection) SetOnInvalidMessageHandler(onInvalidMessageHandler server.OnInvalidMessageHandler) {
 	c.connection.SetOnInvalidMessageHandler(onInvalidMessageHandler)
 }
+
+func (c *NetConnection) setOnDisconnectedHandler(onDisconnectedHandler server.OnDisconnectedHandler) {
```

Comment:
maybe move it to constructor?

Reply:
Impossible, since netAdapter does `delete(na.connections, netConnection)`, a.k.a. needs a reference to the netConnection in the disconnectHandler